### PR TITLE
Merge with LedgerHQ

### PR DESCRIPTION
**Pull request by relatko**
Bump version, add one missing overflow check, rephrase one condition to make it easier to comprehend

### CODE REVIEW

File: app/src/parser_impl.c [hunk 1]
```diff
@@ -199,7 +199,7 @@ parser_error_t json_matchKeyValue(parsed_json_t *parsedJson,
                                   uint16_t tokenIdx, char *expectedType, jsmntype_t jsonType, uint16_t *valueTokenIdx) {
     CHECK_PARSER_ERR(json_validateToken(parsedJson, tokenIdx))
 
-    if (tokenIdx + 4 >= parsedJson->numberOfTokens) {
+    if (! (tokenIdx + 4 < parsedJson->numberOfTokens)) {
```
Comment (ppershing) [inline app/src/parser_impl.c:202]: let's be consistent, there is one extra space here

**Review by ppershing — COMMENTED** [review]
nice

File: app/src/parser_impl.c (hunk 1 above)
Comment (relatko) [inline app/src/parser_impl.c:202]:
```suggestion
    if (!(tokenIdx + 4 < parsedJson->numberOfTokens)) {
```

File: app/src/parser_impl.c [hunk 2]
```diff
@@ -265,7 +265,7 @@ parser_error_t json_matchOptionalArray(parsed_json_t *parsedJson,
                                   uint16_t tokenIdx, uint16_t *valueTokenIdx) {
     CHECK_PARSER_ERR(json_validateToken(parsedJson, tokenIdx))
 
-    if (tokenIdx + 4 >= parsedJson->numberOfTokens) {
+    if (!(tokenIdx + 4 < parsedJson->numberOfTokens)) {
```
Comment (tarakby) [inline app/src/parser_impl.c:268]: Out of curiosity, what is the reason of switching from `<=` to `!>` ?

**Review by tarakby — APPROVED** [review]

File: app/src/parser_impl.c (hunk 2 above)
Comment (relatko) [reply app/src/parser_impl.c:268]: I believe this way it is easier to understand in the context. It forced me to think for quite a while, but I though that I am just tired. Then ledger guys had questions here too.

**Pull request merged**